pkg/handler/handler.go

```diff
@@ -96,26 +100,82 @@ type ReadmeUtgaveFilter struct {
 	First  *int32
 }
 
-func (r *resolver) ReadmeUtgaver(filter *ReadmeUtgaveFilter) ([]ReadmeUtgave, error) {
-	if filter == nil {
-		return r.readmes, nil
-	}
-	var filteredReadmes []ReadmeUtgave
-	for _, r := range r.readmes {
-		if filter.Year != nil && *filter.Year != r.Year {
-			continue
+func getLink(name string) string {
+	return fmt.Sprintf("https://storage.googleapis.com/%v/%v", bucketName, name)
+}
+
+func getReadmes(ctx context.Context, client storage.Client, query string) ([]ReadmeUtgave, error) {
+	pdfQuery := &storage.Query{Prefix: "pdf/" + query}
+	imageQuery := &storage.Query{Prefix: "images/" + query}
+
+	re := regexp.MustCompile(`(?P<Year>\d{4})-(?P<Utgave>\d{2})`)
+
+	bkt := client.Bucket(bucketName)
+	pdfIt := bkt.Objects(ctx, pdfQuery)
+	imageIt := bkt.Objects(ctx, imageQuery)
+	var err error
+	var readmes  []ReadmeUtgave
+	for {
+		pdfAttrs, err := pdfIt.Next()
+		if err == iterator.Done {
+			break
+		}
+		imageAttrs, err := imageIt.Next()
+
+		if err == iterator.Done {
+			break
 		}
-		if filter.Utgave != nil && *filter.Utgave != r.Utgave {
+
+		// Object may be a directory
+		match := re.FindAllString(pdfAttrs.Name, -1)
+		if len(match) == 0 {
 			continue
```

Comment:
Wouldn't this fuck up the ordering of the "double" iterators?

Reply:
It shouldn't :tm:. When iterating, google treats directories and files the same, so an object in the iterator may be a directory, in this case, the object in the image should:tm: also be a directory. I.e. if we have no match, we hit a directory, and we continue.

Of course, this assumes the directory structures of the images and pdfs are **exactly** the same. But this is what you get when you treat a bucket like a database :man_shrugging:

Reply:
Yee, I guess that is a readme problem... A simple JSON would have been way better.🙃🙃 

But is the sorting stable and predictable (eg. sorted by name or date?). (would eg. changing an omd file break the ordering?)

Reply:
It's sorted by name, so should be fine.

Reply:
So both pdf and image has the same filename? Why don't just loop through pdfs, and create image link from the pdf date?

Reply:
That feels way better than this "lets assume they are [xyz]"

Reply:
Hmm, yeah I guess that would be equivalent to the current method. And it might simplify a bit. it does also ensure that we don't get the wrong image :+1:

Reply:
Yee. With the double iterator you risk getting the wrong images for a "utgave", but with a single iterator you just get 404 on the image if there is something :fish:y